de.metas.adempiere.adempiere/base/src/main/java/de/metas/bpartner/service/IBPartnerDAO.java

```diff
@@ -76,12 +76,16 @@
 
 	<T extends I_C_BPartner> T getById(BPartnerId bpartnerId, Class<T> modelClass);
 
-	/** @deprecated Please use {@link IBPartnerDAO#retrieveBPartnerIdBy(BPartnerQuery)} instead.*/
+	/**
+	 * @deprecated Please use {@link IBPartnerDAO#retrieveBPartnerIdBy(BPartnerQuery)} instead.
+	 */
 	@Deprecated
 	Optional<BPartnerId> getBPartnerIdByValue(String bpartnerValue);
 
 	Optional<BPartnerId> getBPartnerIdBySalesPartnerCode(String salesPartnerCode);
 
+	Optional<BPartnerId> getBPartnerIdByExternalId(@NonNull ExternalId externalId);
```

Comment:
please use {@link IBPartnerDAO#retrieveBPartnerIdBy(BPartnerQuery)} isntend (extend and add externalId, if needed)

Reply:
btw, keep in mind that different AD_Orgs might have the same externalId for different bpartners within their respective org.
after all, the externalId's value is controlled not by us, but by some webshop, or similar entitiy